src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceTransformer.java

```diff
@@ -72,15 +70,19 @@
 public class JavaSurfaceTransformer {
   private final GapicCodePathMapper pathMapper;
   private final PackageMetadataConfig packageMetadataConfig;
+
+  // TODO: Simplify by making JavaSurfaceTransformer abstract and making
+  // JavaDiscoGapicSurfaceTransformer and JavaGapicSurfaceTransformer derive from it and override
+  // the methods that are used below.
```

Comment:
No, I don't want to make `JavaSurfaceTransformer` abstract. Such designs don't make the codebase simpler.

Reply:
I disagree. This seems like a case where inheritance would be quite appropriate, and certainly less confusing than both classes referencing each other and doing so by calling the other `surfaceTransformer` (which was the case before before I changed one to be `commonSurfaceTransformer`)

Reply:
This is a conversation about a design choice that doesn't impact the current PR. Since we have disagreement, I would prefer to have a separate conversation about this design choice instead of closing on it within the present PR (which would be required in order to codify your TODO).

Reply:
I agree. That's why I put it as a TODO :-)

If you want, I can alter the wording s/`TODO: Simplify`/`TODO: Consider simplifying`/g

Reply:
Oh, this will be an interesting discussion! I look forward to it.

In the meantime, I'll change the TODO as you suggest.